HARDHAT_DEV_NOTES.md

```diff
@@ -0,0 +1,110 @@
+# periodicRewardsInjector
+
+## Intro
+The ChildChainGaugeInjector is designed to manage weekly injections of non-bal tokens into  Child Chain layer zero gauges based on a predefined keeper using chainlink keepers.
+It is meant to be deployed for a single ERC20 reward token, but can manage any number of gauges for that token.
+
+It allows the configured admin to define a list of schedules, and then can be operated by [Chainlink Automation](https://automation.chain.link/).
+
+Methods exists to add new schedules without implicating the already running schedules for a gauge.
+
+The contract also includes functionality for the owner to sweep ERC20 tokens and gas tokens in order to allow the contract to be easily decommissioned should there be any issues or it is no longer needed.
+
+
+### The Child Chain Gauge runs on weekly epochs:
+
+- Only the defined distributor for a given token may inject rewards.  
+- The injector uses changes to period_finish on the gauge contract to understand epochs and runs once per epoch as early as possible.
```

Comment:
here the reference to `period_finish` will be great to have a reference to docs/contract of what actually is

Reply:
Could perhaps document that, but this is more part of the Balancer Gauge contract.    Would it help to add a sentance here that says, which describes when the current/last set of direct emissions for a given reward token ends/ended?